### PR TITLE
Prevent font element from overriding generic font family (#1042).

### LINKED ISSUE #1042: Clarify that font resources cannot override generic font families.
See https://lists.w3.org/Archives/Public/public-tt/2019Jan/0052.html

### PR DESCRIPTION
**Pull request by skynavga**
Closes #1042.

### CODE REVIEW

**Comment by skynavga** [comment]
@nigelmegitt re: https://github.com/w3c/ttml2/pull/1049#pullrequestreview-215562308, I disagree that we want this to be case sensitive (even though `tts:fontFamily` uses case sensitive matching). The reason we want case insensitive is that we don't want an author to use a downloaded font they label as family `Default` and then use `tts:fontFamily="Default"`, which would match a case sensitive compare and be different from `tts:fontFamily="default"`, which could appear in the same document, meaning match the built-in default font. This could be very confusing to readers of such a document.

**Comment by nigelmegitt** [comment]
@skynavga we're agreeing! To quote myself from https://github.com/w3c/ttml2/pull/1049#pullrequestreview-215562308:

> I think we should adopt case insensitive matching, which would follow CSS

That would require changes elsewhere though.

**Comment by skynavga** [comment]
Actually, we are not agreeing. You are trying to introduce case insensitive matching for family names in general. We cannot do that. However, in the context of the `font` element, we do want case insensitive matching for a different reason entirely, which I have outlined above.

**Comment by skynavga** [comment]
Hmm. I went back and read (carefully) the text defining `<family-name>`, and it appears that one can indeed use `tts:fontFamily="Default"` which is defined mean something different from `tts:fontFamily="default"`, since the former is considered to be a non-generic font family name and the latter is considered to be a generic font family name. That would argue for using case sensitive matching in this PR as you suggest. I will make that change. If you want to file a different issue to change all matches of font family names to be case insensitive, then go ahead and do so; however, we have had case sensitive comparison (only for generic font family matches) ever since TTML1 2e (it was not defined in TTML1 1e).

**Comment by skynavga** [comment]
@nigelmegitt please file a different issue if you want to switch to case-insensitive family comparison across the board; note that it has been case-sensitive since TTML1 2e, so making a change would break backward compatibility in that regard; nothwithstanding, please complete your review and approve so we can move forward on this PR (which the original commenter has already approved)

**Comment by nigelmegitt** [comment]
I requested in a comment on #1042 to add a note to §9.1.4 font saying: 

> If the `family` attribute matches a generic family name then it does not override the generic family of the same name, and cannot be referenced by a `tts:fontFamily` attribute value.

The existing text proposed in this pull request fulfils this purpose already.

**Comment by skynavga** [comment]
@nigelmegitt thanks

**Comment by nigelmegitt** [comment]
@skynavga or @palemieux (or anyone else): Do you have any draft tests associated with the normative changes in this pull request? 

I ask because TTWG agreed that we would try to work with a model in which tests and spec text were prepared together, both to reduce the CR-exit burden later but also to act as a useful check-point that the proposed test and the spec change work together and do what is needed.

Ideally I would like to hold off merging this pull request until we have matching tests.

**Comment by skynavga** [comment]
@nigelmegitt see https://github.com/w3c/ttml2-tests/pull/202

File: spec/ttml2.xml [hunk 1]
```diff
@@ -7434,6 +7434,12 @@ if the specified attribute value(s) differ from the value(s) of these font chara
 the specified attribute value(s) are to be used instead of the font characteristics encoded in the <loc href="#terms-font-resource">font resource</loc>.</p>
 <p>If any of the <att>family</att>, <att>range</att>, <att>style</att>, or <att>weight</att> attributes are not specified, then their values
 must be considered to be equal to the value(s) of the same named font characteristics encoded in the <loc href="#terms-font-resource">font resource</loc>.</p>
+<p>If the computed value of the <code>family</code> attribute (whether explicitly specified or not) matches a
+<loc href="#style-value-generic-family-name">&lt;generic-family-name&gt;</loc> (by case-sensitive comparison),
+then it is considered an error, and the referenced font resource must be ignored for the purpose of presentation processing.</p>
```
[the remark this answers is not in the document]
Comment (skynavga) [reply spec/ttml2.xml]: @nigelmegitt please curtail your propensity to nitpick; what is said is consistent with what we say elsewhere and provides a default behavior for presentation processing; that is enough; must I remind you that other mechanisms take care of whether an error causes and abort or a warning, etc.? we are certainly not going to repeat those semantics here

**Comment by skynavga** [comment]
Tests at w3c/ttml2-tests#202.

**Comment by skynavga** [comment]
Tests at w3c/ttml2-tests#255.

**Pull request merged**